Backends/CLX-fb/port.lisp

```diff
@@ -19,19 +19,16 @@
   (clim-extensions:port-all-font-families port))
 
 (defun initialize-clx-framebuffer (port)
-  (clim-sys:make-process (lambda ()
-                           (loop
-                             (handler-case
-                                 (maphash-keys
-                                  (lambda (key)
-                                    (when (typep key 'clx-fb-mirrored-sheet-mixin)
-                                      (image-mirror-to-x (sheet-mirror key))))
-                                  (slot-value port 'climi::sheet->mirror))
-                               (condition (condition)
-                                 (format *debug-io* "~A~%" condition)))
-                             (xlib:display-force-output (clx-port-display port))
-                             (sleep 0.01)))
-                         :name (format nil "~S's event process." port)))
+  (clim-sys:make-process
+   (lambda ()
+     (loop with sheet->mirror = (slot-value port 'climi::sheet->mirror)
+           do (handler-case
+                  (maphash-values #'image-mirror-to-x sheet->mirror)
+                (condition (condition)
+                  (format *debug-io* "~A~%" condition)))
+              (xlib:display-force-output (clx-port-display port))
+              (sleep 0.01)))
+   :name (format nil "~S's event process." port)))
```

Comment:
I haven't looked in detail, but could the reason for this check be that the graft is also in the hash table?

Reply:
could be, however image-mirror-to-x worked on other objects (by doing nothing) - commit before this one changes the generic function image-mirror-to-x to an ordinary function that has (unless (typep mirror 'clx-fb-mirror) …)

previously it was something like
```
(defgeneric image-mirror-to-x (sheet)
  (:method ((sheet clx-fb-mirror)) (do-things)
  (:method ((sheet t)) #|do-nothing|#))
```

and it was changed before thsi commit to
```
(defun image-mirror-to-x (mirror)
  (unless (typep mirror 'clx-fb-mirror)
    (return-from image-mirror-to-x)) 
  (do-things))
```

Reply:
I still think it would be better to filter unsuitable mirrors here instead of making `image-mirror-to-x` responsible for rejecting random arguments.

Reply:
fixed.